### PR TITLE
Update Filename Validation

### PR DESCRIPTION
**Pull request by Chopinsky**
Associated Issue: #245 

### Summary of Changes

1) Now we will validate the file extension instead of only matching the extension name with the last few characters of the file name.
2) Removing the "." at the end of the error messages, since the errors are displayed inline instead of showing as a standalone sentence.

### Test Plan

Follow the test plan below, and tried to create or edit file name which will now be considered illegal, e.g. for C files, these files won't be allowed for file names: "fileName.cppc", "fileNamec", ".c", etc.

- [x] Create C "hello, world" project
- [x] Click Build button 
- [x] Click Run button, and observe the Result panel
- [x] Clicking the "x" on the Result panel closes it

### CODE REVIEW

File: src/components/EditFileDialog.tsx [hunk 1]
```diff
@@ -48,27 +52,34 @@ export class EditFileDialog extends React.Component<EditFileDialogProps, {
       fileType,
     };
   }
+
```
[the remark this answers is not in the document]
Comment (Chopinsky) [reply src/components/EditFileDialog.tsx]: Removed most of the empty lines unrelated to the changes

File: src/components/NewFileDialog.tsx [hunk 2]
```diff
@@ -46,33 +46,42 @@ export class NewFileDialog extends React.Component<{
...
+      if (fileNameError) {
+        return fileNameError;
+      }
+
+      const directory = this.props.directory;
+      if (directory && appStore.getImmediateChild(directory, this.state.name)) {
+        return `File '${this.state.name}' already exists`;
       }
     }
+
```
Comment (yurydelendik) [inline src/components/NewFileDialog.tsx]: nit: remove unrelated changed such as empty lines or indentation -- it is expected to have separate PR for that.

File: src/components/NewFileDialog.tsx [hunk 3]
```diff
@@ -27,17 +27,17 @@ import appStore from "../stores/AppStore";
 import { File, FileType, Directory, extensionForFileType, nameForFileType, ModelRef, getIconForFileType } from "../models";
 import { ChangeEvent } from "react";
 import { ListBox, ListItem, TextInputBox } from "./Widgets";
+import { FileDialogState } from "./EditFileDialog";
```
Comment (yurydelendik) [inline src/components/NewFileDialog.tsx]: add NewDialogState interface instead of importing it from FileDialogState

**Comment by yurydelendik** [comment]
Thank you for the patch!

**Comment by Chopinsky** [comment]
My pleasure!

**Pull request merged**